fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -2309,6 +2312,20 @@ def _fsdp_instances(self) -> List["FullyShardedDataParallel"]:
         return [m for m in self.modules() if isinstance(m, FullyShardedDataParallel)]
 
     def _remove_uncollectable_params_from_optim_state_dict(self, osd: Dict) -> Dict:
+        """Return a new state dict filtering out the ones like MoE layers, which has
```

Comment:
Where do we need this new function? Is this part of the tests refactoring change or something else?

Reply:
This is not a new function. I added a comment